src/nbl/video/device_features.json

```diff
@@ -0,0 +1,2883 @@
+                "type": "bool",
+                "name": "rayTracingPipelineTraceRaysIndirect",
+                "value": "rayTracingPipeline",
+                "expose": false,
+                "comment": "[DO NOT EXPOSE] Vulkan feature requirements"
+            },
+            {
+                "type": "bool",
+                "name": "rayTraversalPrimitiveCulling",
+                "value": false
+            },
+            {
+                "groupComment": [
+                    "RayQueryFeaturesKHR",
+                    "VK_KHR_ray_query"
+                ]
+            },
+            {
+                "type": "bool",
+                "name": "rayQuery",
+                "value": false
+            },
+            {
+                "groupComment": [
+                    "[DO NOT EXPOSE] just reserved numbers, extension never shipped",
+                    "VK_NV_extension_152",
+                    "[ALIASED TO] VK_AMD_mixed_attachment_samples",
+                    "VK_NV_framebuffer_mixed_samples",
+                    "[DO NOT EXPOSE] For now. For 2D ui",
+                    "VK_NV_fill_rectangle",
+                    "[EXPOSE AS A LIMIT] Enabled by Default, Moved to Limits",
+                    "ShaderSMBuiltinsFeaturesNV",
+                    "VK_NV_shader_sm_builtins",
+                    "Enabled by Default, Moved to Limits",
+                    "VK_EXT_post_depth_coverage",
+                    "[DEPRECATED] Vulkan 1.1 Core and ROADMAP 2022",
+                    "SamplerYcbcrConversionFeaturesKHR",
+                    "VK_KHR_sampler_ycbcr_conversion",
+                    "[DEPRECATED] Core 1.1 implemented on default path and there's no choice in not using it",
+                    "VK_KHR_bind_memory2",
+                    "[DO NOT EXPOSE] Too ",
+                    "intrinsically",
+                    "linux\"\"VK_EXT_image_drm_format_modifier",
+                    "[DO NOT EXPOSE] just reserved numbers, extension never shipped",
+                    "VK_EXT_extension_160",
+                    "[TODO LATER] Expose when we start to experience slowdowns from validation",
+                    "VK_EXT_validation_cache",
+                    "[DEPRECATED] KHR extension supersedes and then Vulkan 1.2",
+                    "VK_EXT_descriptor_indexing",
+                    "[DO NOT EXPOSE] We don't support yet",
+                    "VK_EXT_shader_viewport_index_layer",
+                    "VK_KHR_portability_subset - PROVISINAL/NOTAVAILABLEANYMORE",
+                    "[DO NOT EXPOSE] not implementing or exposing VRS in near or far future, also has interactions with fragment density maps",
+                    "ShadingRateImageFeaturesNV",
+                    "VK_NV_shading_rate_image",
+                    "[DEPRECATED] Superseded by KHR",
+                    "VK_NV_ray_tracing",
+                    "RepresentativeFragmentTestFeaturesNV",
+                    "VK_NV_representative_fragment_test"
```

Comment:
thats broken down in a super weird way, I'd rather have them as hidden variables (no export) with visible comments

Reply:
upvoting

Reply:
Done